src/main/java/com/github/xabgesagtx/bots/TelegramWebhookBotService.java

```diff
@@ -0,0 +1,20 @@
+package com.github.xabgesagtx.bots;
+
+import org.telegram.telegrambots.bots.TelegramWebhookBot;
+import org.telegram.telegrambots.meta.api.methods.updates.SetWebhook;
+
+import javax.validation.constraints.NotNull;
+
+public abstract class TelegramWebhookBotService extends TelegramWebhookBot {
+	/**
+	 * Override this to configure the webhook for a single service here.
+	 * A set of configurations like "certificate" and "external Url" will be preconfigured
+	 *
+	 * @param webhook never null
+	 * @return the modified webhook
+	 */
+	public SetWebhook.SetWebhookBuilder getInitializingWebhookRequest(@NotNull SetWebhook.SetWebhookBuilder webhook) {
```

Comment:
I think the @NotNull is not really doing what you want it to do. The javax.validation annotation need to be validated by someone (some kind of proxy). Do you maybe mean the lombok annotation @NonNull to ensure that this doesn't happen? Or am I misunderstanding something here?
Also, what about naming this `customizeWebHook` or `modifyWebHook` as method name? I had trouble reading/understanding this initially.

Reply:
Good points, thank you! Renamed the method